### ISSUE TITLE
Compiling REFPROP-cmake on macOS 10.14.1 Mojave

### ISSUE THREAD

**Issue by kallovsky** [usnistgov/REFPROP-cmake#22]
I am trying to compile REFPROP on my 2018 MacBook Pro and am running into some problems (please be aware, I am a compiler amateur and just want to use REFPROP with Matlab on a Mac...):

Installed current python, gcc, cmake all via homebrew:

```
$ python --version
Python 2.7.15

$ python -c "import numpy; print(numpy.__version__)"
1.15.4
```

### First problem (when using the instructions):

```
$ cmake --build .

...
[ 53%] Building Fortran object CMakeFiles/refprop.dir/FORTRAN_temp/MIX_AGA8.FOR.o
[ 56%] Building Fortran object CMakeFiles/refprop.dir/FORTRAN_temp/MIX_HMX.FOR.o
[ 60%] Building Fortran object CMakeFiles/refprop.dir/FORTRAN_temp/PASS_FTN.FOR.o
[ 63%] Building Fortran object CMakeFiles/refprop.dir/FORTRAN_temp/PROP_SUB.FOR.o
/Users/MatlabUser/Code/BrenngasGit/TOOLBOX KALIBRIERUNG/Refprop v10/Kompiliere v10/REFPROP-cmake/build/FORTRAN_temp/PROP_SUB.FOR:3681:0:

          hcf=hcf(i+1:255)

Warning: '__builtin_memmove' reading 255 bytes from a region of size 254 [-Wstringop-overflow=]
[ 66%] Building Fortran object CMakeFiles/refprop.dir/FORTRAN_temp/REALGAS.FOR.o
[ 70%] Building Fortran object CMakeFiles/refprop.dir/FORTRAN_temp/REFPROP.FOR.o
/Users/MatlabUser/Code/BrenngasGit/TOOLBOX KALIBRIERUNG/Refprop v10/Kompiliere v10/REFPROP-cmake/build/FORTRAN_temp/REFPROP.FOR:484: Error: Can't open included file 'COMMONS.INC'
make[2]: *** [CMakeFiles/refprop.dir/FORTRAN_temp/REFPROP.FOR.o] Error 1
make[1]: *** [CMakeFiles/refprop.dir/all] Error 2
make: *** [all] Error 2
```

Obviously, it can't find COMMONS.INC (and also the other .INC-Files)
=> If I copy those .INC files explicitly into FORTRAN_temp/ it compiles further, until:

### Second problem:

The compilation comes to a halt with lots of ld-errors ending in:

```
...
 "_xsqrt_", referenced from:
     _allprops0_ in REFPROP.FOR.o
     _convunits_ in REFPROP.FOR.o
ld: symbol(s) not found for architecture x86_64
collect2: error: ld returned 1 exit status
make[2]: *** [librefprop.dylib] Error 1
make[1]: *** [CMakeFiles/refprop.dir/all] Error 2
make: *** [all] Error 2
MacBookPro:build MatlabUser$
```

I have no idea how to recover from this - it would be great if someone could have a look!

Thanks for your help,

Carsten

**Comment by ianhbell** [comment]
Can you please provide the *entire* build log? There's some potentially important information about what command line flags you used, what compilers were located, etc. that are missing

**Comment by kallovsky** [comment]
Sorry for the late reply - here's the complete commandline output (that's what you mean, right?):
```
MacBookUSER:REFPROP-cmake USER$ ls
Build-Log.txt		FORTRAN			docker			generate_aliases.py
CMakeLists.txt		README.md		externals		testRP.m
MacBookUSER:REFPROP-cmake USER$ mkdir build
MacBookUSER:REFPROP-cmake USER$ cd build
MacBookUSER:build USER$ cmake .. -DCMAKE_BUILD_TYPE=Release
-- The C compiler identification is AppleClang 10.0.0.10001044
-- The CXX compiler identification is AppleClang 10.0.0.10001044
-- Check for working C compiler: /Library/Developer/CommandLineTools/usr/bin/cc
-- Check for working C compiler: /Library/Developer/CommandLineTools/usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /Library/Developer/CommandLineTools/usr/bin/c++
-- Check for working CXX compiler: /Library/Developer/CommandLineTools/usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- The Fortran compiler identification is GNU 8.2.0
-- Checking whether Fortran compiler has -isysroot
-- Checking whether Fortran compiler has -isysroot - yes
-- Checking whether Fortran compiler supports OSX deployment target flag
-- Checking whether Fortran compiler supports OSX deployment target flag - yes
-- Check for working Fortran compiler: /usr/local/bin/gfortran
-- Check for working Fortran compiler: /usr/local/bin/gfortran  -- works
-- Detecting Fortran compiler ABI info
-- Detecting Fortran compiler ABI info - done
-- Checking whether /usr/local/bin/gfortran supports Fortran 90
-- Checking whether /usr/local/bin/gfortran supports Fortran 90 -- yes
-- Found PythonInterp: /usr/local/bin/python (found version "2.7.15")
-- DEFSYM_FLAG:
-- darwin
Missing from PASS_CMN_tokens:  ['ABFLSHdll', 'ALLPROPS0dll', 'ALLPROPS1dll', 'ALLPROPS20dll', 'ALLPROPSdll', 'CRTPNTdll', 'DEFL1dll', 'DERVPVTdll', 'DHFL1dll', 'DSD1dll', 'ERRMSGdll', 'FLAGSdll', 'GERG08dll', 'GETENUMdll', 'GETMODdll', 'GETREFDIRdll', 'HEATFRMdll', 'HMXORDERdll', 'HSFL1dll', 'MASSFLUXdll', 'MELTKdll', 'PASSCMNdll', 'PEFL1dll', 'PHI0dll', 'PHIDERVdll', 'PHIHMXdll', 'PHIKdll', 'PHIMIXdll', 'RDXHMXdll', 'REFPROP1dll', 'REFPROP2dll', 'REFPROPdll', 'RIEMdll', 'SATESTdll', 'SATGUESSdll', 'SETFLUIDSdll', 'SETMIXTUREdll', 'SETREFDIRdll', 'STNdll', 'TEFL1dll', 'THFL1dll', 'TSATPdll', 'TSFL1dll', 'VIRBCDdll', 'WMOLIdll']
Missing from REFPROP_lib.h: ['DPDDKdll', 'DPDTKdll', 'ACTVYdll', 'DOTFILLdll', 'OMEGAdll', 'ETAK0dll', 'ETAK1dll', 'ETAKRdll', 'ETAKBdll', 'TCXK0dll', 'TCXKBdll', 'TCXKCdll']

-- Path to libquadmath.a:/usr/local/Cellar/gcc/8.2.0/lib/gcc/8/gcc/x86_64-apple-darwin18.0.0/8.2.0/../../../libquadmath.a
-- Copied libquadmath.a here.
-- REFPROP version <= 9.1 identified by presence of include "'commons.for'"; copying locally and fixing include statement
-- Fixing of FORTRAN files complete. Compilation ready.
-- Set REFPROP_FORTRAN_PATH to /Users/USER/Code/BrenngasGit/TOOLBOX KALIBRIERUNG/Refprop v10/Kompiliere v10/REFPROP-cmake/build/FORTRAN_temp
-- Configuring done
-- Generating done
-- Build files have been written to: /Users/USER/Code/BrenngasGit/TOOLBOX KALIBRIERUNG/Refprop v10/Kompiliere v10/REFPROP-cmake/build
MacBookUSER:build USER$ cmake --build .
Scanning dependencies of target REFPROP_H
[  3%] About to build the REFPROP.h header file w/ /usr/local/bin/python;-u;/Users/USER/Code/BrenngasGit/TOOLBOX KALIBRIERUNG/Refprop v10/Kompiliere v10/REFPROP-cmake/externals/REFPROP-headers/generate_header.py;--FORTRAN-path;/Users/USER/Code/BrenngasGit/TOOLBOX KALIBRIERUNG/Refprop v10/Kompiliere v10/REFPROP-cmake/build/FORTRAN_temp;--python-exe;/usr/local/bin/python
Writing the .pyf file with numpy.f2py, please be patient...
Deleting REFPROP.pyf
[  3%] Built target REFPROP_H
Scanning dependencies of target refprop
[  6%] Building Fortran object CMakeFiles/refprop.dir/FORTRAN_temp/CORE_ANC.FOR.o
[ 10%] Building Fortran object CMakeFiles/refprop.dir/FORTRAN_temp/CORE_BWR.FOR.o
[ 13%] Building Fortran object CMakeFiles/refprop.dir/FORTRAN_temp/CORE_CPP.FOR.o
[ 16%] Building Fortran object CMakeFiles/refprop.dir/FORTRAN_temp/CORE_DE.FOR.o
[ 20%] Building Fortran object CMakeFiles/refprop.dir/FORTRAN_temp/CORE_ECS.FOR.o
[ 23%] Building Fortran object CMakeFiles/refprop.dir/FORTRAN_temp/CORE_FEQ.FOR.o
[ 26%] Building Fortran object CMakeFiles/refprop.dir/FORTRAN_temp/CORE_MLT.FOR.o
[ 30%] Building Fortran object CMakeFiles/refprop.dir/FORTRAN_temp/CORE_PH0.FOR.o
[ 33%] Building Fortran object CMakeFiles/refprop.dir/FORTRAN_temp/CORE_PR.FOR.o
[ 36%] Building Fortran object CMakeFiles/refprop.dir/FORTRAN_temp/CORE_QUI.FOR.o
[ 40%] Building Fortran object CMakeFiles/refprop.dir/FORTRAN_temp/CORE_STN.FOR.o
[ 43%] Building Fortran object CMakeFiles/refprop.dir/FORTRAN_temp/FLASH2.FOR.o
[ 46%] Building Fortran object CMakeFiles/refprop.dir/FORTRAN_temp/FLSH_SUB.FOR.o
[ 50%] Building Fortran object CMakeFiles/refprop.dir/FORTRAN_temp/IDEALGAS.FOR.o
[ 53%] Building Fortran object CMakeFiles/refprop.dir/FORTRAN_temp/MIX_AGA8.FOR.o
[ 56%] Building Fortran object CMakeFiles/refprop.dir/FORTRAN_temp/MIX_HMX.FOR.o
[ 60%] Building Fortran object CMakeFiles/refprop.dir/FORTRAN_temp/PASS_FTN.FOR.o
[ 63%] Building Fortran object CMakeFiles/refprop.dir/FORTRAN_temp/PROP_SUB.FOR.o
[ 66%] Building Fortran object CMakeFiles/refprop.dir/FORTRAN_temp/REALGAS.FOR.o
[ 70%] Building Fortran object CMakeFiles/refprop.dir/FORTRAN_temp/REFPROP.FOR.o
/Users/USER/Code/BrenngasGit/TOOLBOX KALIBRIERUNG/Refprop v10/Kompiliere v10/REFPROP-cmake/build/FORTRAN_temp/REFPROP.FOR:484: Error: Can't open included file 'COMMONS.INC'
make[2]: *** [CMakeFiles/refprop.dir/FORTRAN_temp/REFPROP.FOR.o] Error 1
make[1]: *** [CMakeFiles/refprop.dir/all] Error 2
make: *** [all] Error 2
```
I manually copy all .INC-Files into FORTRAN_temp/ and try again:
```
MacBookUSER:build USER$ cmake --build .
[  3%] Built target REFPROP_H
[  6%] Building Fortran object CMakeFiles/refprop.dir/FORTRAN_temp/REFPROP.FOR.o
/Users/USER/Code/BrenngasGit/TOOLBOX KALIBRIERUNG/Refprop v10/Kompiliere v10/REFPROP-cmake/build/FORTRAN_temp/REFPROP.FOR:5589:72:

          if (ABS(iErrPrnt).eq.3) pause           !If your compiler complains about this line, just delete it.
                                                                        1
Warning: Deleted feature: PAUSE statement at (1)
[ 10%] Building Fortran object CMakeFiles/refprop.dir/FORTRAN_temp/SAT_SUB.FOR.o
[ 13%] Building Fortran object CMakeFiles/refprop.dir/FORTRAN_temp/SETUP.FOR.o
[ 16%] Building Fortran object CMakeFiles/refprop.dir/FORTRAN_temp/SETUP2.FOR.o
[ 20%] Building Fortran object CMakeFiles/refprop.dir/FORTRAN_temp/TRNSP.FOR.o
[ 23%] Building Fortran object CMakeFiles/refprop.dir/FORTRAN_temp/TRNS_ECS.FOR.o
[ 26%] Building Fortran object CMakeFiles/refprop.dir/FORTRAN_temp/TRNS_TCX.FOR.o
[ 30%] Building Fortran object CMakeFiles/refprop.dir/FORTRAN_temp/TRNS_VIS.FOR.o
[ 33%] Building Fortran object CMakeFiles/refprop.dir/FORTRAN_temp/UTILITY.FOR.o
[ 36%] Linking Fortran shared library librefprop.dylib
Undefined symbols for architecture x86_64:
  "_aqflsh_", referenced from:
      _abflsh_ in REFPROP.FOR.o
  "_changeall_", referenced from:
      _refset1_ in REFPROP.FOR.o
  "_checklimits_", referenced from:
      _refprop_ in REFPROP.FOR.o
  "_dsd1_", referenced from:
      _allprops0_ in REFPROP.FOR.o
  "_errnum2_", referenced from:
      _errmsg_ in REFPROP.FOR.o
  "_errnum_", referenced from:
      _abflsh_ in REFPROP.FOR.o
      _convunit_ in REFPROP.FOR.o
      _getenum_ in REFPROP.FOR.o
      _flags_ in REFPROP.FOR.o
      _setfluids_ in REFPROP.FOR.o
      _readfluidstring_ in REFPROP.FOR.o
      _fldsetup_ in REFPROP.FOR.o
      ...
  "_exp5_", referenced from:
      _refset1_ in REFPROP.FOR.o
  "_gerg08_", referenced from:
      _flags_ in REFPROP.FOR.o
  "_getinputs_", referenced from:
      _abflsh_ in REFPROP.FOR.o
  "_getprops_", referenced from:
      _abflsh_ in REFPROP.FOR.o
      _refprop_ in REFPROP.FOR.o
  "_getqq_", referenced from:
      _abflsh_ in REFPROP.FOR.o
  "_heatfrm_", referenced from:
      _allprops0_ in REFPROP.FOR.o
  "_hvapztn_", referenced from:
      _allprops0_ in REFPROP.FOR.o
  "_idhconv_", referenced from:
      _errmsg_ in REFPROP.FOR.o
  "_leq0_", referenced from:
      _abflsh_ in REFPROP.FOR.o
      _flags_ in REFPROP.FOR.o
      _readfluidstring_ in REFPROP.FOR.o
      _allprops0_ in REFPROP.FOR.o
  "_lequal_", referenced from:
      _refsav_ in REFPROP.FOR.o
      _allprops0_ in REFPROP.FOR.o
  "_ltrim_", referenced from:
      _refset2_ in REFPROP.FOR.o
      _setmixture_ in REFPROP.FOR.o
      _allprops_ in REFPROP.FOR.o
  "_satguess_", referenced from:
      _abflsh_ in REFPROP.FOR.o
  "_stringsplit_", referenced from:
      _setfluids_ in REFPROP.FOR.o
  "_swaperr_", referenced from:
      _abflsh_ in REFPROP.FOR.o
      _setmixture_ in REFPROP.FOR.o
      _fldsetup_ in REFPROP.FOR.o
      _allprops0_ in REFPROP.FOR.o
      _allprops_ in REFPROP.FOR.o
      _refprop_ in REFPROP.FOR.o
  "_tpdflsh_", referenced from:
      _abflsh_ in REFPROP.FOR.o
  "_ucase10000_", referenced from:
      _readfluidstring_ in REFPROP.FOR.o
      _allprops_ in REFPROP.FOR.o
  "_virbcd12_", referenced from:
      _allprops0_ in REFPROP.FOR.o
  "_xdiv3_", referenced from:
      _allprops0_ in REFPROP.FOR.o
      _allprops_ in REFPROP.FOR.o
  "_xdiv_", referenced from:
      _convunit_ in REFPROP.FOR.o
      _getunits_ in REFPROP.FOR.o
      _allprops0_ in REFPROP.FOR.o
      _refprop_ in REFPROP.FOR.o
  "_xsqrt_", referenced from:
      _allprops0_ in REFPROP.FOR.o
      _convunits_ in REFPROP.FOR.o
ld: symbol(s) not found for architecture x86_64
collect2: error: ld returned 1 exit status
make[2]: *** [librefprop.dylib] Error 1
make[1]: *** [CMakeFiles/refprop.dir/all] Error 2
make: *** [all] Error 2
MacBookUSER:build USER$
```

**Comment by ianhbell** [comment]
What version of REFPROP are you are using?  The presence of the line saying that it detected a version <= 9.1 is concerning.  Are you trying to compile version 8?  Anything above version 9 should work fine

**Comment by kallovsky** [comment]
Sorry! Obviously, I copied an older version of the FORTRAN-directory. I actually wanted to compile version 10...

The compilation runs without major problems now - with only one exception:
```
MacBookUSER:build USER$ cmake --build .
Scanning dependencies of target REFPROP_H
make[2]: *** No rule to make target `../fortran/PASS_FTN.FOR', needed by `REFPROP.h.stamp'.  Stop.
make[1]: *** [CMakeFiles/REFPROP_H.dir/all] Error 2
make: *** [all] Error 2
MacBookUSER:build USER$
```
PASS_FTN.FOR is in a sub-directory, so if I copy that file:
```
MacBookUSER:build USER$ cp ../FORTRAN/DLLFILES/PASS_FTN.FOR ../FORTRAN/
```
everything is fine!

Sorry again for that stupid mistake, and thanks for your help!

**Comment by ianhbell** [comment]
Hmm....  I thought I had fixed that bug with the DLLFILES thingy.  Are you sure you pulled the most recent version of REFPROP-cmake?

**Comment by kallovsky** [comment]
Yes, I downloaded the most recent version right before compiling!

**Comment by ianhbell** [comment]
Should have been fixed in 152f1c6dd3cecbb866b1287253ca5de822e75e4e

**Comment by kallovsky** [comment]
Just checked out that version – and everything is compiling smoothly now!

Thanks for your help!
